SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -17702,24 +17715,36 @@ void CNSSolver::Source_Residual(CGeometry *geometry, CSolver **solver_container,
     for (unsigned long iPoint = 0; iPoint < nPointDomain; iPoint++) {
 
       const su2double Volume = geometry->node[iPoint]->GetVolume();
-      const su2double* U = node[iPoint]->GetSolution();
+
+      // Use mean density in forcing, to eliminate (as much as
+      // possible) effect of correlation between rho and Force
+      const su2double* U = average_node[iPoint]->GetSolution();
 
       const su2double* Force = HybridMediator->GetForcingVector(iPoint);
+      const su2double* MeanForce = average_node[iPoint]->GetForce();
 
       // Set forcing in variable class (for output purposes)
       node[iPoint]->SetForcingVector(Force);
+      average_node[iPoint]->SetForcingVector(MeanForce);
 
       // mass (no forcing)
       Residual[0] = 0.0;
 
       // momentum
-      for (unsigned short iDim = 0; iDim < nDim; iDim++)
-        Residual[iDim+1] = -Volume * U[0] * Force[iDim];
+      for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+        //Residual[iDim+1] = -Volume * U[0] * Force[iDim];
+	Residual[iDim+1] = -Volume * U[0] * (Force[iDim] - MeanForce[iDim]);
+	//Residual[iDim+1] = -Volume * (Force[iDim] - MeanForce[iDim]);
+      }
 
       // energy
       Residual[nDim+1] = 0.0;
-      for (unsigned short iDim = 0; iDim < nDim; iDim++)
-        Residual[nDim+1] += -Volume * U[iDim+1] * Force[iDim];
+      for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+        //Residual[nDim+1] += -Volume * U[iDim+1] * Force[iDim];
+	//Residual[nDim+1] += -Volume * U[iDim+1] * (Force[iDim] - MeanForce[iDim]);
+	Residual[nDim+1] += 0.0; // no source in energy?
+	//Residual[nDim+1] += -Volume * (U[iDim+1]/U[0]) * (Force[iDim] - MeanForce[iDim]);
+      }
```

Comment:
@trevilo What's going on here?

Reply:
Ha, just me being dumb.  I originally had the forcing entering the total energy equation, since that is what you'd do for a "normal" forcing term (i.e., a source in the momentum equation).  But, the forcing here is a transfer from unresolved to resolved.  Therefore, there isn't any forcing contribution to the total energy equation.